### ISSUE TITLE
Assess the difficulty of changing Mathigon UI from a dev perspective

### ISSUE THREAD

**Issue by delapuente** [Qiskit/qiskit.org#929]
Part of #782 

Explore official or unofficial ways of changing/extending Mathigon UI. Propose an example of how would it be to adapt Mathigon UI to our style, based on current design mocks. If there is no official way to do it, think of an extension mechanism enabling us to implement our requirements with reuse in mind.

- [x] #935
- [x] #938

**Comment by delapuente** [comment]
Dup of #930

**Comment by vabarbosa** [comment]
the recommended/documented approach for updating the theme/UI in Mathigon includes:

- updating a course's primary color and/or main icon via the course's [`curriculum.yaml`](https://github.com/mathigon/textbooks/blob/master/curriculum.yaml)
- style changes (using LESS) to specific course content (e.g., [`styles.less`](https://github.com/mathigon/textbooks/blob/master/content/circles/styles.less))
- adding ids, classes and attributes to paragraphs or inline elements via custom markdown format

however to provide more in-depth changes to the layout & UI would require making changes at the server level. the core styles and templates are stored and served by the server. with access to these files a more complete rewrite of the theme/UI would be made possible.

access to the server at this time is not available. there is a [`@mathigon/dev-server`](https://github.com/mathigon/dev-server) which gives some insight on the server and can be used for testing/demoing. however, it does not cover all aspects of the production server.

with that being said, defining a customizable theme/UI is possible with some initial changes/updates to the server. the customization, configuration, styles, etc. could then be defined alongside the course. this would allow for different courses to provide different look-and-feel and override the default UI provided by the server.

in summary, as it currently works Mathigon does provide some basic UI customization, but for more complete theming options (e.g., layout changes, template replacements, styling overhaul, etc.) additional access/work beyond just creating the content is required. while not complicated it will take time and resources to update the server and create the desired themes/UI content to align with the desired look & feel and still adhere to Mathigon's approach.

**Comment by delapuente** [comment]
Thank you for your assessment.
